diff --git a/include/vuk/RenderGraph.hpp b/include/vuk/RenderGraph.hpp
index 59dc785..8c9645b 100644
--- a/include/vuk/RenderGraph.hpp
+++ b/include/vuk/RenderGraph.hpp
@@ -21,7 +21,7 @@
 #include <unordered_set>
 #include <vector>
 
-#if false // seems like clang still has issues with this - lets keep it safe
+#if true
 #define VUK_IA(access, ...)        vuk::Arg<vuk::ImageAttachment, access, decltype([]() {}) __VA_OPT__(, ) __VA_ARGS__>
 #define VUK_BA(access, ...)        vuk::Arg<vuk::Buffer, access, decltype([]() {}) __VA_OPT__(, ) __VA_ARGS__>
 #define VUK_ARG(type, access, ...) vuk::Arg<type, access, decltype([]() {}) __VA_OPT__(, ) __VA_ARGS__>
diff --git a/include/vuk/runtime/CommandBuffer.hpp b/include/vuk/runtime/CommandBuffer.hpp
index 1a7347a..3aa751b 100644
--- a/include/vuk/runtime/CommandBuffer.hpp
+++ b/include/vuk/runtime/CommandBuffer.hpp
@@ -358,9 +358,7 @@ namespace vuk {
 		DomainFlagBits get_scheduled_domain() const;
 
 		/// @brief Gets the allocator associated with this object.
-		Allocator get_allocator() {
-			return *allocator;
-		}
+		Allocator get_allocator();
 
 		// command buffer state setting
 		// when a state is set it is persistent for a pass (similar to Vulkan dynamic state) - see documentation
diff --git a/src/runtime/vk/CommandBuffer.cpp b/src/runtime/vk/CommandBuffer.cpp
index 21747fc..a54a778 100644
--- a/src/runtime/vk/CommandBuffer.cpp
+++ b/src/runtime/vk/CommandBuffer.cpp
@@ -72,6 +72,10 @@ namespace vuk {
 		return stream->domain;
 	}
 
+	Allocator CommandBuffer::get_allocator() {
+		return *allocator;
+	}
+
 	CommandBuffer& CommandBuffer::set_descriptor_set_strategy(DescriptorSetStrategyFlags ds_strategy_flags) {
 		this->ds_strategy_flags = ds_strategy_flags;
 		return *this;
